uber/config.py

```diff
@@ -72,6 +72,72 @@ def PREREG_BADGE_TYPES(self):
         return types
 
     @property
+    def BADGE_DISPLAY_TYPES(self):
+        """
+        There are several contexts where we want to display different badge types to select:
+        - A new attendee registering for the event
+        - An attendee claiming a blank badge in an existing group
+        - An existing attendee editing their registration
+
+        This property uses the parameters in the URL to figure out the context and return the correct
+        badge types to display for an attendee.
+        """
+        with sa.Session() as session:
+            params = cherrypy.lib.httputil.parse_query_string(cherrypy.request.query_string)
+
+            # This isn't ideal, but bypasses cases where "id" refers to a group id.
+            try:
+                group = session.group(params['group_id']) if 'group_id' in params else None
+                attendee = session.attendee(params['id']) if 'id' in params else None
+            except:
+                group = None
+                attendee = None
+
+            if not attendee:
+                # Inherits the group's badge type if it's an attendee claiming a badge in a group
+                base_badge_name = group.ribbon_and_or_badge if group else c.BADGES.get(c.ATTENDEE_BADGE)
+            else:
+                base_badge_name = attendee.ribbon_and_or_badge
+
+            donation_prepend = '' if base_badge_name == c.BADGES.get(c.ATTENDEE_BADGE) else base_badge_name + ' / '
+            badge_cost = attendee.badge_cost if attendee else c.BADGE_PRICE
+
+            # The base badge and the group badge are special, so they're given default values and added manually
+            base_description = 'Allows access to the convention for its duration.'
+
+            if c.BEFORE_GROUP_PREREG_TAKEDOWN:
+                group_description = 'Register a group of ' + str(c.MIN_GROUP_SIZE) + ' or more and save $' + \
+                                    str(c.GROUP_DISCOUNT) + ' per badge.'
+            else:
```

Comment:
Fun fact; Python concatenates strings by default, so in cases where you want to split a string into parts and put each part on a line like this, you don't even need the plus and can literally just say

```python
foo = ('Hello '
       'World')
```
or if you prefer
```python
foo = 'Hello ' \
      'World'
```

I'm personally more of a "just put it all on one long line" kind of guy, but that's just a style preference and our PEP8 checker has no preference :)